File: gitdot-api/src/endpoint/runner/list_runners.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{RunnerResource, common::Page},
};

pub struct ListRunners;

impl Endpoint for ListRunners {
    const PATH: &'static str = "/ci/runner/{owner}";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListRunnersRequest;
    type Response = ListRunnersResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListRunnersRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListRunnersResponse = Page<RunnerResource>;

File: gitdot-web/packages/gitdot-api/src/endpoint/runner/list-runners.ts
import { z } from "zod";
import { page, RunnerResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListRunnersRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListRunnersRequest = z.infer<typeof ListRunnersRequest>;

export const ListRunnersResponse = page(RunnerResource);
export type ListRunnersResponse = z.infer<typeof ListRunnersResponse>;

export const ListRunners = {
  path: "/ci/runner/{owner}",
  method: "GET",
  request: ListRunnersRequest,
  response: ListRunnersResponse,
} as const satisfies Endpoint;
export type ListRunners = typeof ListRunners;
